building-docs-in-pull-requests.rst

```diff
@@ -0,0 +1,116 @@
+
+Google Summer of Code is a global program where students work with an open source organization
+on a 3 month programming project. **Autobuild Documentation in Pull Requests** was one of the project ideas on
+Read the Docs `Project Ideas`_ for Google Summer of Code 2019. I applied for the project and got accepted.
+
+I started working on this project since May 2019.
+All of my work on this project can be seen in the `Pull Request Builder Board`_.
+
+Background
+==========
+
+Many of our users wanted a way to visualize the documentation update that was made in a pull request.
+They also want to know whether the documentation build will pass before merging the pull request to master.
+This would allow users to have more confidence on the pull request
+and make the pull request less likely to break the documentation after merging.
+So, to achieve this Read the Docs `core team`_ selected **Autobuild docs for Pull Requests**
+as one of the projects of Google Summer of Code 2019.
+
+Pull Request Builder Features
+=============================
+
+Currently I have implemented these features and working on more upcoming features.
+Some of the major features are as following:
+
+- **Creating External Versions:** We create an external version when we receive a pull request
+  webhook event for a project from GitHub and trigger a build for that version.
+  External versions are short-lived versions for a Read the Docs project that are separate from the project's main documentation.
+
+- **Synchronizing External Versions:** Whenever there is a new commit on the pull request,
+  we synchronize the external version for that pull request and trigger a new documentation build with the latest changes.
+
+- **Deleting External Versions:** Whenever the pull request is closed or merged,
+  we delete the external Version associated with that pull request.
+
+- **Warning Banner for pull request documentation:** While building documentation for pull requests
+  we add a warning banner at the top of those documentations to let the users know that
+  this documentation was generated for pull requests and is not the main documentation for the project.
+
+We send build status reports to status API of the provider (e.g. GitHub).
+When a build is triggered for a pull request we send build pending notification with the build URL
+and after the build has finished we send success notification if the build succeeded without any error
+or failure notification if the build failed.
+By going to the build URL provided in the status report users can view the build steps
+and also see the documentation generated by that build.
+
+**Currently, we only support GitHub and planning to extend to GitLab and BitBucket**
+
+Getting Started
+===============
+
+Building documentation for pull requests is currently in Beta testing and only supports Github repositories.
+If you want to dive in and enable this feature for your project you can `email us <mailto:support@readthedocs.org>`_.
```

Comment:
We should mention that people should include a project URL that they want to enable.